strutil/shlex/shlex.go

```diff
@@ -414,3 +415,28 @@ func Split(s string) ([]string, error) {
 		subStrings = append(subStrings, word)
 	}
 }
+
+// Join returns a shell-escaped string version of the slice s
+func Join(s []string) string {
+	var quoted []string
```

Comment:
I think you can create a slice of a known size here, to reduce the number of possible allocations during append (when it will grow)
```go
quoted := make([]string, len(s))
```

And then in the `for-loop` just set the value of `quoted[i]`

Reply:
Thank you. Done.